WalletWasabi/Crypto/GroupElementVector.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Crypto
+{
+	public class GroupElementVector : IEnumerable<GroupElement>
+	{
+		public GroupElementVector(IEnumerable<GroupElement> groupElements)
+		{
+			GroupElements = groupElements.ToArray();
+		}
+
+		private IEnumerable<GroupElement> GroupElements { get; }
+
+		public IEnumerator<GroupElement> GetEnumerator() =>
+			GroupElements.GetEnumerator();
+
+		IEnumerator IEnumerable.GetEnumerator() =>
+			GetEnumerator();
+
+		public int Count => GroupElements.Count();
+
+		public GroupElement Sum() =>
+			GroupElements.Aggregate(GroupElement.Infinity, (ge, acc) => ge + acc);
+
+		public GroupElement InnerProduct(ScalarVector scalars)
```

Comment:
FWIW if this is renamed to just an `*` overload, this should *only* support component wise multiplication and summing, not something, i.e. `ScalarVector * GroupElementVector`, and not `Scalar * GroupElementVector`, to avoid any possibility of e.g. unintentionally using a single secret nonce multiple times instead of several distinct ones, or something like that

Reply:
I think I am wrong and it is not an inner product even it looks like one. I will convert it to and `*` operator.

Reply:
yes. it is confusing, because it is not an inner product. typically in an inner product, you would have the same objects. Here you have a list of scalars and a list of points. So as discussed above, I agree with just overleading the * operator.